sound/soc/sof/topology.c

```diff
@@ -1856,17 +1824,39 @@ static int sof_process_load(struct snd_soc_component *scomp, int index,
 		if (pdata->magic != SOF_ABI_MAGIC)
 			continue;
 
-		memcpy(&process->data + offset, pdata->data, pdata->size);
-		offset += pdata->size;
+		switch (scontrol->cmd) {
+		case SOF_CTRL_CMD_VOLUME:
+		case SOF_CTRL_CMD_ENUM:
+		case SOF_CTRL_CMD_SWITCH:
+			ipc_cmd = SOF_IPC_COMP_SET_VALUE;
+			ctrl_type = SOF_CTRL_TYPE_VALUE_CHAN_SET;
+			break;
```

Comment:
@juimonen I'm not sure about this. We cant pass any private data in the kcontrol as a volume set command. Either there has to be a sanity check to make sure it is in the volume table or we remove it. I think it is unnecessary to pass any other data that the binary control data.

Reply:
ok